src/workflows/set-password.ts

```diff
@@ -9,11 +9,16 @@ type SetPasswordOutput = {
   details?: PasswordRules;
 };
 
-async function setPassword(passwordInput: string): Promise<SetPasswordOutput> {
+async function setPassword(
+  passwordInput: string,
+): Promise<
+  | { isUpdated: boolean }
+  | { code: string; message: string; details?: PasswordRules }
+> {
   return fetchJson("/api/auth-connect/set-password", HttpVerbs.POST, {
     passwordInput,
   }).then((response) => {
-    if (response.status >= 400 && response.status !== 422) {
+    if (response.status > 400 && response.status !== 422) {
```

Comment:
Why not include `400`s?

Reply:
I excluded 400s because if I don't, the invalidBody webError get catch there and triggers an unhandled error (the one we had Sentry report yesterday). I followed what we did in `addIdentity` workflow : 

```typescript
if (response.status > 400) {
      throw new ErrorSendingValidationCode();
}
```

But I really wonder if we should keep these code blocks in the workflows. If the `ConnectUnreachable` webError is raised, it will get catched inside them too.

Reply:
I agree we should remove this workflow 👍